docs/specifications/linking-and-embedding-specification.md

```diff
@@ -0,0 +1,364 @@
+
+The rules governing the word-based linking are as follows:
+
+- Pointing to a note is done via its H1 heading, e.g. `[[My Note]]`.
+- The word-based wikilink in converted to a normal markdown link using the link reference definitions. Example:
+  - `[Some Title]: some-file.md`
+- In case the workspace contains two or more notes with the same heading, disambiguation is done with the link reference definitions [1]. Example:
+  - `[Some Title]: some-file.md`
+  - `[Some Title:2]: subdirectory/some-file.md`
+- In case the note heading is missing, an arbitrary link can be used, which is then retargeted with the link reference definitions: `[Some Arbitrary Label]: some-file-without-heading.md` [1]
+- Pointing to a heading within a note is done using the heading as an anchor: `[Some File#Some Heading]` [2]
+- If a note contains multiple headings with the same text, link reference definitions must disambiguate between them using the running number of the identifier, e.g.: `[Some File#Some Heading]: some-file.md#some-heading-1`.
+- Pointing to an arbitrary anchor within a note is done via its id (e.g. `[[Some File#foo]]`), since anchors might not a have a title (e.g. `<a id="foo"></a>`) [2].
+
+[1]: This implies that the link reference definitions are not ephemeral, and they cannot be reconstructed solely based on the index.
+[2]: This, in its full form (e.g. including the anchor), is recorded into the link reference definitions.
+
+##### Path-based Wikilinks
+
+Path-based wikilinks point to the file names, and if the note is in subfolders in relation to the workspace root, these folders are also included directly into the link itself. The benefit of the path-based links is that they are more unambiguous than the word-based links, and they also expose the potential semantics of the directory structure. They also generate smaller amount of link reference definitions, thus reducing the need for programmatic insertion of text into the files. The drawback is that they look more like "code", and might seem foreign especially to non-coder types.
+
+The rules governing the path-based linking are as follows:
+
+- Pointing to a note is done via its workspace rooted path, e.g. `[[some/path/here.md]]`
+- Links can also be written without the file extensions, like `[[some/path/here]]`
+- In case there's conflicts due to lack of file extensions (e.g. `here.md` and `here.doc` in the same path), link reference definitions are used to disambiguate (e.g. `[some-file]: some-file.md`) OR the file extension is added into the link itself OR an error is shown to the user.
+- Pointing to a heading within a note is done based on id, e.g. `[[foo/bar#my-title]]`
+- Pointing to an arbitrary anchor within a note is done also based on id, e.g. `[[foo/bar#my-anchor]]`
+
+##### Link Labels
+
+All wikilinks can have an optional label assigned to them. The assignment is done by separating the link target and the label with a pipe-character: `[[Some Note#Some Anchor|Some Label]]`
```

Comment:
I wonder if the additional syntax here is necessary, or whether the labeling could be done as the "Title" part of the LinkRefDef?

What is the purpose of labels in work-based wikilinks? Why not use the arbitrary word content as the label?

In path-based links I see the usefulness of it, as it avoids having to generate a LinkRefDef while still allowing for a custom display text.

I'm not opposed to this syntax, just seeking to understand the use case.